experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1338,82 +1314,121 @@ def validate_hypothesis(self, value):
             raise serializers.ValidationError("Hypothesis cannot be the default value.")
         return value
 
-    def _validate_feature_value_against_schema(self, schema, value):
-        json_value = json.loads(value)
-        try:
-            jsonschema.validate(json_value, schema, resolver=NestedRefResolver(schema))
-        except jsonschema.ValidationError as exc:
-            return [exc.message]
+    def _validate_feature_value_against_schema(self, feature_config, value):
+        if feature_config.schema:
+            schema = json.loads(feature_config.schema)
+
+            json_value = json.loads(value)
+            try:
+                jsonschema.validate(
+                    json_value, schema, resolver=NestedRefResolver(schema)
+                )
+            except jsonschema.ValidationError as exc:
+                return [exc.message]
 
     def _validate_feature_configs(self, data):
+        errors = {}
+
         feature_configs = data.get("feature_configs", [])
+        if len(feature_configs) <= 1:
+            return data
+
+        warn_feature_schema = data.get("warn_feature_schema", False)
 
         for feature_config in feature_configs:
             if self.instance.application != feature_config.application:
-                raise serializers.ValidationError(
-                    {
-                        "feature_configs": [
-                            f"Feature Config application {feature_config.application} "
-                            f"does not match experiment application "
-                            f"{self.instance.application}."
-                        ]
-                    }
-                )
+                errors["feature_configs"] = [
+                    f"Feature Config application {feature_config.application} "
+                    f"does not match experiment application "
+                    f"{self.instance.application}."
+                ]
+
+        reference_branch_errors = []
+        for feature_value_data in data.get("reference_branch", {}).get(
+            "feature_values", []
+        ):
+            if schema_errors := self._validate_feature_value_against_schema(
+                feature_value_data["feature_config"], feature_value_data["value"]
+            ):
+                reference_branch_errors.append({"value": schema_errors})
+                continue
+
+            reference_branch_errors.append({})
```

Comment:
no issues with the code, just continue is disturbing me :D we can do something like this too 
```python
reference_branch_errors = [{"value": self._validate_feature_value_against_schema(feature_value_data["feature_config"], feature_value_data["value"]
                        )} if (schema_errors:=self._validate_feature_value_against_schema(feature_value_data["feature_config"], feature_value_data["value"])) else {} for feature_value_data in data.get("reference_branch", {}).get("feature_values", [])]

Reply:
Yeah I'm not a huge fan of continue either, it's just that otherwise you need multiple duplicate else cases and that also bothered me so I considered this the lesser of two evils.

And yes in theory you can write it the way you've suggested, but as little a fan of continue as I am, I'm much significantly less a fan of using very long single expressions.  They're harder to read, reason about, and modify, so generally I try to stick to 'a single logical operation per line' as much as possible.  But thank you for the suggestion!